tests/lib/fakeinstaller/main.go

```diff
@@ -0,0 +1,282 @@
+		if stru.Label == "" || stru.Filesystem == "" {
+			continue
+		}
+
+		part, err := disk.FindMatchingPartitionWithPartLabel(stru.Label)
+		if err != nil {
+			return nil, err
+		}
+		// XXX: reuse
+		// gadget/install/content.go:mountFilesystem() instead
+		// (it will also call udevadm)
+		if err := mkfs.Make(stru.Filesystem, part.KernelDeviceNode, stru.Label, 0, 0); err != nil {
+			return nil, err
+		}
+
+		// mount
+		mountPoint := runMntFor(stru.Label)
+		if err := os.MkdirAll(mountPoint, 0755); err != nil {
+			return nil, err
+		}
+		// XXX: is there a better way?
+		if output, err := exec.Command("mount", part.KernelDeviceNode, mountPoint).CombinedOutput(); err != nil {
+			return nil, osutil.OutputErr(output, err)
+		}
+		mountPoints = append(mountPoints, mountPoint)
+	}
+
+	return mountPoints, nil
+}
+
+func unmountFilesystems(mntPts []string) error {
+	for _, mntPt := range mntPts {
+		if output, err := exec.Command("umount", mntPt).CombinedOutput(); err != nil {
+			return osutil.OutputErr(output, err)
+		}
+	}
+	return nil
+}
+
+func createClassicRootfsIfNeeded(rootfsCreator string) error {
+	dst := runMntFor("ubuntu-data")
+
+	if output, err := exec.Command(rootfsCreator, dst).CombinedOutput(); err != nil {
+		return osutil.OutputErr(output, err)
+	}
+
+	return nil
+}
+
+func createSeedOnTarget(bootDevice, seedLabel string) error {
+	// XXX: too naive?
+	dataMnt := runMntFor("ubuntu-data")
+	src := dirs.SnapSeedDir
+	dst := dirs.SnapSeedDirUnder(dataMnt)
+	if err := os.MkdirAll(filepath.Dir(dst), 0755); err != nil {
+		return err
+	}
+	if output, err := exec.Command("cp", "-a", src, dst).CombinedOutput(); err != nil {
+		return osutil.OutputErr(output, err)
+	}
+
+	return nil
+}
+
+func run(seedLabel, bootDevice, rootfsCreator string) error {
+	if len(os.Args) != 4 {
+		// xxx: allow installing real UC without a classic-rootfs later
+		return fmt.Errorf("need seed-label, target-device and classic-rootfs as argument\n")
+	}
+	os.Setenv("SNAPD_DEBUG", "1")
+	logger.SimpleSetup()
+
+	logger.Noticef("installing on %q", bootDevice)
+
+	cli := client.New(nil)
+	details, err := cli.SystemDetails(seedLabel)
+	if err != nil {
+		return err
+	}
+	// TODO: grow the data-partition based on disk size
+	laidoutStructs, err := createPartitions(bootDevice, details.Volumes)
+	if err != nil {
+		return fmt.Errorf("cannot setup partitions: %v", err)
+	}
+	if err := postSystemsInstallSetupStorageEncryption(details); err != nil {
+		return fmt.Errorf("cannot setup storage encryption: %v", err)
+	}
+	mntPts, err := createAndMountFilesystems(bootDevice, details.Volumes)
+	if err != nil {
+		return fmt.Errorf("cannot create filesystems: %v", err)
+	}
+	if err := createClassicRootfsIfNeeded(rootfsCreator); err != nil {
+		return fmt.Errorf("cannot create classic rootfs: %v", err)
+	}
+	if err := createSeedOnTarget(bootDevice, seedLabel); err != nil {
+		return fmt.Errorf("cannot create seed on target: %v", err)
+	}
+	// Unmount filesystems
```

Comment:
(nitpick) the comment seems to not add much over the `umountFilesystems()` call so we might as well remove it maybe?